home_work/hw_7/like/likes.py

```diff
@@ -0,0 +1,31 @@
+"""Module for the Likes class implementation."""
+
+
+class Likes:
+    @staticmethod
+    def likes(*names: str | list[str]) -> str:
+        if all(isinstance(name, (str, list)) for name in names) == 0:
+            raise TypeError("Names must be strings or lists of strings")
+
+        names_list: list[str] = []
+        for name in names:
+            if isinstance(name, str):
+                names_list.append(name)
+            else:
+                names_list.extend(name)
+
+        num_likes = len(names_list)
+
+        if num_likes == 0:
+            return "no one likes this"
+
+        if num_likes == 1:
```

Comment:
fix

Reply:
> We do not want to execute all `if ` statements, because of time and resources, so it is better to use: `if-elif-else`.

It won't enter each if because it will stop at the first return.